### PR TITLE
Make it an error to use a class-attribute type var outside a type

### PR DESCRIPTION
**Pull request by sixolet**
Previously, e9d28a0 fixed a crash when you tried to access a class-attribute
type variable. The test in that commit involved assigning a variable the value
of the typevar. It resulted in no crash, but rather treating the variable as
being an instance of the type the typevar bound to, later, which is incorrect.

For example (and this is the test defined in e9d28a0, modified)

```python
 from typing import TypeVar
 class C:
     T = TypeVar('T', int)
     def f(self, x: T) -> T:
         A = C.T
         reveal_type(A)  # E: revealed type is "T`-1"
         return x
```

Instead, this PR treats such an assignment as an error -- any reference to such a classvar tvar outside trying to construct a type out of it is an error.

Also test a *correct* alias with the typevar access method in question -- it
works.

### CODE REVIEW

File: mypy/typeanal.py [hunk 1]
```diff
@@ -43,6 +43,9 @@ def analyze_type_alias(node: Expression,
     # that we don't support straight string literals as type aliases
     # (only string literals within index expressions).
     if isinstance(node, RefExpr):
+        # Note that this misses the case where someone tried to use a
+        # class-referenced type variable as a type alias.  It's easier to catch
+        # that one in checkmemeber.py
```
Comment (JelleZijlstra) [inline mypy/typeanal.py]: small typo: checkmember.py

Comment (sixolet) [inline mypy/typeanal.py]: meme-ber.  Thank you.

**Comment by sixolet** [comment]
I'm not exactly sure about the wording of the error here, suggestions welcome.

File: test-data/unit/check-typevar-values.test [hunk 2]
```diff
@@ -496,12 +496,18 @@ def outer(x: T) -> T:
...
 
 [case testClassMemberTypeVarInFunctionBody]
-from typing import TypeVar
+from typing import TypeVar, List
 class C:
     T = TypeVar('T', bound=int)
     def f(self, x: T) -> T:
-        A = C.T
-        return x
+        L = List[C.T] # a valid type alias
```
Comment (JelleZijlstra) [inline test-data/unit/check-typevar-values.test]: Would doing `y: C.T = x` be valid?

Comment (sixolet) [inline test-data/unit/check-typevar-values.test]: Yes, and good point -- that's not "as a parameter to a type" that's "as a type"

How to express "Don't try and use this at runtime"?

Comment (sixolet) [inline test-data/unit/check-typevar-values.test]: I guess "Is only valid when used as a type or a parameter to a type in a type alias" covers all the bases, but is wicked confusing if I don't already know exactly what it's talking about.

Comment (ilevkivskyi) [inline test-data/unit/check-typevar-values.test]: Some time ago we had "... is invalid in runtime context" for subscripted generics. Maybe it is better to say "... can not be used in runtime context, only in type context".

Comment (sixolet) [inline test-data/unit/check-typevar-values.test]: Oh!  How about 
* "Type variable 'C.T' is not a valid expression"
* "Type variable 'C.T' cannot be used as an expression"
?

Comment (ilevkivskyi) [inline test-data/unit/check-typevar-values.test]: The second option sounds good.

Comment (sixolet) [inline test-data/unit/check-typevar-values.test]: Ok, rolling with @ilevkivskyi's suggestion for now.

Comment (sixolet) [inline test-data/unit/check-typevar-values.test]: ... as I incur more and more test changes as I can't decide.  I *think* I like how it is now.

**Comment by JukkaL** [comment]
Thanks for the PR! LGTM

**Pull request merged**